File: src/my_project_typescript/easy_problems_typescript/from51To100/destinationCity.ts
function destCity(paths: string[][]): string {
    let originList: Array<string> = []
    let destinationList: Array<string> = []


    paths.map( currentElement => {
        originList.push(currentElement[0])
    })

    paths.map(city => {
        if (!originList.includes(city[1])){
            destinationList.push(city[1])
        }
    })

    return destinationList[0]
};

console.log(destCity([["London","New York"],["New York","Lima"],["Lima","Sao Paulo"]]))

console.log(destCity([["B","C"],["D","B"],["C","A"]]))

console.log(destCity([["A","Z"]]))

File: src/my_project/easy_problems/from151To200/destination_city.py
from typing import List 

class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        origin_list: List[str] = list()
        destination_list: List[str] = list()

        for city in paths:
            origin_list.append(city[0])

        
        for city in paths:
            if city[1] not in origin_list:
                destination_list.append(city[1])

        return destination_list[0]

solution = Solution()

print(solution.destCity(paths = [["London","New York"],["New York","Lima"],["Lima","Sao Paulo"]]))

print(solution.destCity(paths = [["B","C"],["D","B"],["C","A"]]))

print(solution.destCity(paths = [["A","Z"]]))

